File: FullTextSearch/src/test/java/SetOperators/OrSetOperatorTest.java
package SetOperators;

import Model.InvertedIndex;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.util.*;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

class OrSetOperatorTest {
    private static Map<String, Set<String>> indexMap;

    @BeforeAll
    public static void initialize() {
        indexMap = new HashMap<>();
        Set<String> helloIndex = new HashSet<>(Arrays.asList("1", "2", "3", "4"));
        Set<String> worldIndex = new HashSet<>(Arrays.asList("2", "5"));
        Set<String> hiIndex = new HashSet<>(Arrays.asList("2", "4"));
        indexMap.put("hello", helloIndex);
        indexMap.put("world", worldIndex);
        indexMap.put("hi", hiIndex);
    }

    @Test
    public void notEmptyInitialSetCommonElementTest() {
        Set<String> result = new HashSet<>(Arrays.asList("a", "b", "c"));
        Set<String> orMask = new HashSet<>(Arrays.asList("a", "b"));
        new OrSetOperator().specificOperation(result, orMask);
        org.assertj.core.api.Assertions.assertThat(result).containsOnly("a", "b", "c");
    }

    @Test
    public void notEmptyInitialSetNonCommonElementTest() {
        Set<String> result = new HashSet<>(Arrays.asList("a", "b", "c"));
        Set<String> orMask = new HashSet<>(Arrays.asList("d", "e"));
        new OrSetOperator().specificOperation(result, orMask);
        org.assertj.core.api.Assertions.assertThat(result).containsOnly("a", "b", "c", "d", "e");
    }

    @Test
    public void EmptyInitialSetTest() {
        Set<String> result = new HashSet<>();
        Set<String> orMask = new HashSet<>(Arrays.asList("z", "y"));
        new OrSetOperator().specificOperation(result, orMask);
        org.assertj.core.api.Assertions.assertThat(result).containsOnly("z", "y");
    }

    @Test
    public void testFullFunctionalityOfOrOnInvertedIndex() {
        InvertedIndex invertedIndex = mock(InvertedIndex.class);

        when(invertedIndex.getIndices()).thenReturn(indexMap);

        Set<String> initial = indexMap.get("hello");
        Set<String> result= new OrSetOperator().operate(initial , Arrays.asList("hi" , "world") , invertedIndex);
        org.assertj.core.api.Assertions.assertThat(result).containsOnly("1","2","3","4","5");


    }
}

File: FullTextSearchCsharp/FullTextSearch/FullTextSearch.Test/SetOperators/OrSetOperatorTest.cs
using System.Collections.Generic;
using FullTextSearch.SetOperators;
using NFluent;
using Xunit;


namespace FullTextSearch.Test.SetOperators
{
    public class OrSetOperatorTest : SetOperatorTest
    {

        [Fact]
        public void NotEmptyInitialSetCommonElementTest()
        {
            ISet<string> result = new HashSet<string> { "a", "b", "c" };
            ISet<string> subMask = new HashSet<string> {"a" , "b" };

            new OrSetOperator().SpecificOperation(result, subMask);
            Check.That(result).IsEquivalentTo("a", "b", "c");
        }


        [Fact]
        public void NotEmptyInitialSetNonCommonElementTest()
        {
            ISet<string> result = new HashSet<string> { "a", "b", "c" };
            ISet<string> subMask = new HashSet<string> { "d" , "e"};

            new OrSetOperator().SpecificOperation(result, subMask);
            Check.That(result).IsEquivalentTo("a", "b", "c" , "d" , "e");
        }


        [Fact]
        public void EmptyInitialSetTest()
        {
            ISet<string> result = new HashSet<string>();
            ISet<string> subMask = new HashSet<string> {"z" , "y" };

            new OrSetOperator().SpecificOperation(result, subMask);
            Check.That(result).IsEquivalentTo("z" , "y");
        }


        [Fact]
        public void TestFullFunctionalityOfOrOnInvertedIndexWithHello()
        {
            var initial = IndexMap["hello"];
            var result =
                new OrSetOperator().Operate(initial, new List<string> { "hi", "world" }, Indexes);
            Check.That(result).IsEquivalentTo("1" , "2" , "3" , "4");

        }

        [Fact]
        public void TestFullFunctionalityOfOrOnInvertedIndexWithHi()
        {
            var initial = IndexMap["hi"];
            var result =
                new OrSetOperator().Operate(initial, new List<string> { "world"}, Indexes);
            Check.That(result).IsEquivalentTo( "2", "3", "4");


        }

        [Fact]
        public void TestFullFunctionalityOfOrOnInvertedIndexWithNullQueryList()
        {
            var initial = IndexMap["hello"];
            var result =
                new OrSetOperator().Operate(initial, null, Indexes);
            Check.That(result).IsEquivalentTo("1", "2", "3", "4");
        }

    }
}